File: src/main/java/com/smarthealthcare/backend/entity/PrescriptionMedicine.java
package com.smarthealthcare.backend.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "prescription_medicines")
public class PrescriptionMedicine {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String medicineName;

    private String strength;

    @Column(columnDefinition = "TEXT")
    private String instruction;

    private Boolean verified = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "prescription_id")
    private Prescription prescription;

    public PrescriptionMedicine() {
    }

    public Long getId() {
        return id;
    }

    public String getMedicineName() {
        return medicineName;
    }

    public void setMedicineName(String medicineName) {
        this.medicineName = medicineName;
    }

    public String getStrength() {
        return strength;
    }

    public void setStrength(String strength) {
        this.strength = strength;
    }

    public String getInstruction() {
        return instruction;
    }

    public void setInstruction(String instruction) {
        this.instruction = instruction;
    }

    public Boolean getVerified() {
        return verified;
    }

    public void setVerified(Boolean verified) {
        this.verified = verified;
    }

    public Prescription getPrescription() {
        return prescription;
    }

    public void setPrescription(Prescription prescription) {
        this.prescription = prescription;
    }
}

File: frontend/lib/models/prescription/prescription_medicine.dart
import 'database_medicine.dart';

class PrescriptionMedicine {
  final String medicineName;
  final String strength;
  final String instruction;
  final bool verified;
  final DatabaseMedicine? databaseMedicine;

  PrescriptionMedicine({
    required this.medicineName,
    required this.strength,
    required this.instruction,
    required this.verified,
    this.databaseMedicine,
  });

  factory PrescriptionMedicine.fromJson(
      Map<String, dynamic> json) {
    return PrescriptionMedicine(
      medicineName: json["medicineName"] ?? "",
      strength: json["strength"] ?? "",
      instruction: json["instruction"] ?? "",
      verified: json["verified"] ?? false,
      databaseMedicine:
          json["databaseMedicine"] == null
              ? null
              : DatabaseMedicine.fromJson(
                  json["databaseMedicine"],
                ),
    );
  }
}